moveit_simple/include/moveit_simple/moveit_simple.h

```diff
@@ -353,15 +341,68 @@ class Robot
   tf2_ros::TransformListener tf_listener_;
 
   // ROS objects
-  ros::NodeHandle nh_;
-  ros::Subscriber j_state_sub_;
   mutable std::recursive_mutex m_;
 
 };
 
 
 
 
+/**
+ * @brief OnlineRobot is a wrapper around standard MoveIt objects.
+ It inherits from Robot. Added assumption:
+ - motion is executed via a "JointTrajectoryAction" interface
+*/
+
+class OnlineRobot : public Robot
+{
+public:
+  OnlineRobot(const ros::NodeHandle & nh, const std::string &robot_description,
+        const std::string &group_name);
+
+  /**
+   * @brief execute a given trajectory
+   * @param traj_name - name of trajectory to be executed (must be filled with
+   * prior calls to "addTrajPoint".
+   * @param collision_check - bool to turn check for collision on\off
+   * @throws <moveit_simple::ExecutionFailureException> (Execution failure)
+   * @throws <moveit_simple::IKFailException> (Conversion to joint trajectory failed)
+   * @throws <std::invalid_argument> (Trajectory "traj_name" not found)
+   * @throws <moveit_simple::CollisionDetected> (One of interpolated point is
+   * in Collision with itself or environment)
+   */
+  void execute(const std::string traj_name, bool collision_check = false);
+
+
+  /**
+  * @brief getJointState - Returns a vector<double> of the
+  * current joint positions of the robot from current_robot_state_.
+  *
+  * @return std::vector<double> current_joint_position
+  */
+  virtual std::vector<double> getJointState(void) const;
+
+protected:
+  OnlineRobot();
+
+  void updateCurrentState(const sensor_msgs::JointStateConstPtr& msg);
+  mutable moveit::core::RobotStatePtr current_robot_state_;
+
+
+  // Visualizations
+  moveit_visual_tools::MoveItVisualToolsPtr visual_tools_;
```

Comment:
Node handles are singletons...no matter how many you create, there is only ever one.  It's safe to the use the same one for both.